packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_splash/views/splash_page.dart

```diff
@@ -0,0 +1,49 @@
+{{> licence.dart }}
+
+import 'package:flutter/material.dart';
+import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
+
+import '../../app_extensions.dart';
+import '../../base/common_ui_components/app_error_card_component.dart';
+import '../../base/common_ui_components/app_loading_indicator.dart';
+import '../../base/extensions/async_snapshot_extensions.dart';
+import '../../base/models/errors/error_model.dart';
+import '../blocs/splash_bloc.dart';
+
+class SplashPage extends StatelessWidget {
+  const SplashPage({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+    body: Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Expanded(
+          child: RxBlocBuilder<SplashBlocType, bool>(
+            state: (bloc) => bloc.states.isLoading,
+            builder: (state, loading, bloc) => loading.isLoading
+                ? Center(
+              child: AppLoadingIndicator.taskValue(
+                context,
+                color: context.designSystem.colors.primaryColor,
+              ),
+            )
+                : RxBlocBuilder<SplashBlocType, ErrorModel>(
```

Comment:
Is there a reason the `RxBlocBuilder<SplashBlocType, ErrorModel>` to be a nested so deep in the widget tree as well as placed using the conditional UI approach. What do you think of placing this widget as a child of the Column, the same way as with the RxBlocBuilder<SplashBlocType, bool>

Reply:
reworked